django/utils/functional.py

```diff
@@ -396,3 +397,24 @@ def partition(predicate, values):
     for item in values:
         results[predicate(item)].append(item)
     return results
+
+
+def batches(iterable, size):
```

Comment:
If size can be `None`, I suggest to make it an optional parameter like `def batches(iterable, size=None):`